.github/workflows/test.yaml

```diff
@@ -13,10 +13,10 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
       - name: Setup Go
-        uses: actions/setup-go@v3
+        uses: actions/setup-go@v5
```

Comment:
Updated to match the go-version-file example here: https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file.

Also updated other files so actions use consistent versions.